server/src/helpers.rs

```diff
@@ -59,42 +61,76 @@ pub fn get_auth_headers(
     }
 }
 
+fn origin(url: &str) -> String {
```

Comment:
does the `actix_web::http:Uri` not provide this with the [`authority` method](https://docs.rs/actix-web/latest/actix_web/http/uri/struct.Uri.html#method.authority)?

Might handle some extra edge cases, I'm not sure.

Reply:
That doesn't include the scheme, so it's not a valid uri (which is the type of the subject field afaict).